### ISSUE TITLE
Community -> space migrations don't account for rate limits

### ISSUE THREAD

**Issue by refi64** [vector-im/element-web#19252]
### Steps to reproduce

- Attempt to migrate a community with over ~10 users to a space, while on a homeserver with the default Synapse ratelimimt configuration.
- Element will just say "Failed to migrate community", with `M_UNKNOWN: Cannot invite so many users at once` shown in the logs.

### What happened?

In my particular case, the community has 41 members. All of the members at once are passed straight to the `create_room` endpoint, which then [ratelimits the invites](https://github.com/matrix-org/synapse/blob/428174f90249ec50f977b5ef5c5cf9f92599ee0a/synapse/handlers/room.py#L712-L723) via a [relatively restrictive default configuration](https://github.com/matrix-org/synapse/blob/524b8ead778e51adfd6667a33f2700f8e071c256/synapse/config/ratelimiting.py#L128-L135):

```python
self.rc_invites_per_room = RateLimitConfig(
    config.get("rc_invites", {}).get("per_room", {}),
    defaults={"per_second": 0.3, "burst_count": 10},
)
self.rc_invites_per_user = RateLimitConfig(
    config.get("rc_invites", {}).get("per_user", {}),
    defaults={"per_second": 0.003, "burst_count": 5},
)
```

Given that the community has 41 members, 5 is evidently far too low to work...

I'm guessing maybe this should create it with the only user being the one that is requesting the migration? Then it can add the others in small groups at a time, with some sort of retry backoff or something to avoid hitting the rate limit. It's still a bit racy, but the current approach of just...not working isn't particularly great. :sweat_smile: 

### Operating system

Fedora Silverblue 35

### Browser information

94.0.4606.71

### URL for webapp

https://element.linuxcafe.chat/

### Homeserver

linuxcafe.chat

### Have you submitted a rageshake?

No

**Comment by refi64** [comment]
I did also get

```
Access to XMLHttpRequest at 'https://linuxcafe.chat/_matrix/client/r0/createRoom' from origin 'https://element.linuxcafe.chat' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

after a "successful" migration which resulted in Element still saying it failed...but this is probably a local configuration issue :sweat_smile: just thought it's worth noting here anyway in case it's related somehow

**Comment by markvoortman** [comment]
See #19216.

**Comment by SimonBrandner** [comment]
It seems you're not the only one to have come across this bug, as it's been reported previously. I'm going to close this issue now so we can focus on the older issue. Please watch, vote or comment on that issue if there is any additional information you can provide.

Duplicate of https://github.com/vector-im/element-web/issues/19216

(please correct me if I am wrong)

**Comment by refi64** [comment]
Apologies for the dup! I was searching "ratelimiting" which didn't find the issue 😅

-- Ryan
https://refi64.com/

**Comment by SimonBrandner** [comment]
> Apologies for the dup! I was searching "ratelimiting" which didn't find the issue 
> […](#)
> -- Ryan [refi64.com](https://refi64.com/)
> On Oct 2, 2021, 1:21 AM -0500, Šimon Brandner ***@***.***>, wrote: Closed #19252. —You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

No worries!
